src/argsparser.c

```diff
@@ -301,6 +302,20 @@ void parse_recurring_delay(char *arg)
     free(x);
 }
 
+int isnumber(char *pin)
```

Comment:
i think this shouldn't be "isnumber" but rather "is_valid_pin". there's a function `wps_pin_checksum()` which you could use to check if the checksum is valid (after checking that the pin is numeric and of the right length). if the pin has an invalid checksum, one-shot mode needs to be activated as well.